### PR TITLE
Add flow definition file

### PR DESCRIPTION
**Pull request by koddsson**
Added a flow definition file.

https://github.com/github/web-systems/issues/215

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -60,77 +64,80 @@ export default class AutoCheckElement extends HTMLElement {
...
+    throw new Error('missing csrf')
+  }
+  const input = autoCheckElement.input
+  if (!input) return
 
-    const body = new FormData()
-    body.append('authenticity_token', this.csrf)
-    body.append('value', this.input.value)
+  const body = new FormData()
+  body.append('authenticity_token', autoCheckElement.csrf)
```
Comment (mislav) [inline src/index.js]: Since `.csrf` is computed programatically via getter, could we save it to a const within this method so we don't access it more than once? Same for `.src`, which is even more expensive since it creates an element

File: src/index.js [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
-export default class AutoCheckElement extends HTMLElement {
-  boundCheck: () => {}
-  input: HTMLInputElement
+let boundCheck
 
+export default class AutoCheckElement extends HTMLElement {
   constructor() {
     super()
-    this.boundCheck = debounce(this.check.bind(this), 300)
+    boundCheck = debounce(check.bind(null, this), 300)
```
[the remark this answers is not in the document]
Comment (koddsson) [reply src/index.js]: I think people have a lot of opinions about the public API, could this we keep these in a `WeakMap` so we have each check on a per instance but keep this out of the public API?

File: src/index.js.flow [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  declare export default class AutoCheckElement extends HTMLElement {
+    get src(): string;
+    set src(value: string): void;
+    get csrf(): string;
+    set csrf(value: string): void;
+    get required(): boolean;
+    set required(required: boolean): void;
+    get input(): ?HTMLInputElement;
+  }
+}
```
Comment (dgraham) [inline src/index.js.flow]: Can you separate the flow definitions into another pull request so that can merge? Coupling it to code changes means it can't merge until those pass code review.

File: src/index.js.flow [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/* flow strict */
+
+declare module 'auto-check-element' {
+  declare export default class AutoCheckElement extends HTMLElement {
+    boundCheck: () => void;
+    input: HTMLInputElement;
```
Comment (dgraham) [inline src/index.js.flow]: These two entries are not supported public API so they can be removed from the type refs here. If we want to support accessing the managed input element, `input` could be defined as a getter.

File: src/index.js.flow [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+    get src(): string;
+    set src(value: string): void;
+    get csrf(): string;
+    set csrf(value: string): void;
+    get required(): boolean;
+    set required(required: boolean): void;
+
+    check(): void;
```
Comment (dgraham) [inline src/index.js.flow]: Is `check` intended to be public API? When does the host app call this directly versus firing a change event at the input?

File: src/index.js.flow (hunk 4 above)
Comment (koddsson) [reply src/index.js.flow]: Yup! That's what I essentially did in this PR but then split that out into #14. I really wanted to represent the code in the repo and currently `input`, `check` and `boundCheck` are accessible via the element.

Omitting properties that are accessible from the flow definition seems a bit off to me but I don't mind removing these in this PR since I do move those properties out the element in #14.
